cmd/gctsExecuteABAPUnitTests.go

```diff
@@ -0,0 +1,365 @@
+
+	var xmlBody = []byte(`<?xml version="1.0" encoding="UTF-8"?>
+	<aunit:runConfiguration
+			xmlns:aunit="http://www.sap.com/adt/aunit">
+			<external>
+					<coverage active="false"/>
+			</external>
+			<options>
+					<uriType value="semantic"/>
+					<testDeterminationStrategy sameProgram="true" assignedTests="false" appendAssignedTestsPreview="true"/>
+					<testRiskLevels harmless="true" dangerous="true" critical="true"/>
+					<testDurations short="true" medium="true" long="true"/>
+			</options>
+			<adtcore:objectSets
+					xmlns:adtcore="http://www.sap.com/adt/core">
+					<objectSet kind="inclusive">
+							<adtcore:objectReferences>
+									<adtcore:objectReference adtcore:uri="/sap/bc/adt/packages/` + packageName + `"/>
+							</adtcore:objectReferences>
+					</objectSet>
+			</adtcore:objectSets>
+	</aunit:runConfiguration>`)
+
+	url := config.Host +
+		"/sap/bc/adt/abapunit/testruns?sap-client=" + config.Client
+
+	resp, httpErr := client.SendRequest("POST", url, bytes.NewBuffer(xmlBody), header, nil)
+
+	defer func() {
+		if resp != nil && resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	if httpErr != nil {
+		return errors.Wrap(httpErr, "execution of unit tests failed")
+	} else if resp == nil {
+		return errors.New("execution of unit tests failed: did not retrieve a HTTP response")
+	}
+
+	var response runResult
+	parsingErr := parseHTTPResponseBodyXML(resp, &response)
+	if parsingErr != nil {
+		log.Entry().Warning(parsingErr)
+	}
+
+	aunitError := parseAUnitResponse(&response)
+	if aunitError != nil {
+		return aunitError
+	}
+
+	return nil
+}
+
+func parseAUnitResponse(response *runResult) error {
+	var node string
+	aunitError := false
+
+	for _, program := range response.Program {
+		log.Entry().Infof("testing class %v", program.Name)
+		for _, testClass := range program.TestClasses.TestClass {
+			log.Entry().Infof("using test class %v", testClass.Name)
+			for _, testMethod := range testClass.TestMethods.TestMethod {
+				node = testMethod.Name
+				if len(testMethod.Alerts.Alert) > 0 {
+					log.Entry().Errorf("%v - error", node)
+					aunitError = true
+				} else {
+					log.Entry().Infof("%v - ok", node)
+				}
+			}
+		}
+	}
+	if aunitError {
+		return errors.Errorf("some unit tests failed")
+	}
+	return nil
+}
+
+func getPackageList(config *gctsExecuteABAPUnitTestsOptions, telemetryData *telemetry.CustomData, client piperhttp.Sender) ([]string, error) {
+
+	type object struct {
+		Pgmid       string `json:"pgmid"`
+		Object      string `json:"object"`
+		Type        string `json:"type"`
+		Description string `json:"description"`
+	}
+
+	type objectsResponseBody struct {
+		Objects   []object      `json:"objects"`
+		Log       []gctsLogs    `json:"log"`
+		Exception gctsException `json:"exception"`
+		ErrorLogs []gctsLogs    `json:"errorLog"`
+	}
+
+	url := config.Host +
+		"/sap/bc/cts_abapvcs/repository/" + config.Repository +
```

Comment:
What about defining the API endpoints as constants to have them all at the beginning of the file?

Reply:
If there is no big reason to do this I would prefer to keep it the way it is.